includes/connect/class-wc-stripe-connect-api.php

```diff
@@ -91,6 +93,11 @@ protected function request( $method, $path, $body = [] ) {
 			}
 
 			$url = trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL );
+
+			if ( isset( $body['mode'] ) && 'test' === $body['mode'] ) {
+				$url = str_replace( 'api.woocommerce.com', 'api-staging.woocommerce.com', $url ); // TODO: replace this once we know the format of a test request to api.woocommerce.com.
+			}
+
```

Comment:
@diegocurbelo I'm thinking that once we have updated the `api.woocommerce.com` Stripe oauth-init endpoint to accept test mode requests we can remove this. We just need to work out if `api.woocommerce.com` will receive a `mode =>"test"|"live"` or `test => true` type param in the request body.

Reply:
To be consistent with how other services are implemented on the connect server, we would need to use a different url, with the same payload: `/stripe-sandbox/oauth-init` and `/stripe-sandbox/oauth-keys`

Draft PR here: https://github.com/Automattic/woocommerce-connect-server/pull/2545